azure/services/aso/aso.go

```diff
@@ -170,6 +180,17 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.ASOReso
 		annotations[ReconcilePolicyAnnotation] = ReconcilePolicyManage
 	}
 
+	// Set the secret name annotation if any of the following conditions are met:
+	//- AzureClusterIdentity name is not empty
+	//- SubscriptionID is different from the one controller is running with
+	// This is needed in order to leverage the ASO resource credential scope as defined in
+	//https://azure.github.io/azure-service-operator/guide/authentication/credential-scope/#resource-scope.
+	if s.azureClusterIdentityName != "" {
+		annotations[SecretNameAnnotation] = aso.GetASOSecretName(s.azureClusterIdentityName, s.subscriptionID)
+	} else if s.subscriptionID != os.Getenv(auth.SubscriptionID) {
+		annotations[SecretNameAnnotation] = aso.GetASOSecretName("", s.subscriptionID)
```

Comment:
this will result in a secret name starting with `-` if I'm reading the `GetASOSecretName` function right

Reply:
Ups, forgot to add the `s.clusterName`, will add it in the next commit